pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -231,28 +213,54 @@ type ReconcileIntrusionDetection struct {
 	licenseAPIReady *utils.ReadyFlag
 	dpiAPIReady     *utils.ReadyFlag
 	tierWatchReady  *utils.ReadyFlag
-	usePSP          bool
+	multiTenant     bool
 	elasticExternal bool
 }
 
+func GetIntrusionDetection(ctx context.Context, cli client.Client, mt bool, ns string) (*operatorv1.IntrusionDetection, error) {
+	key := client.ObjectKey{Name: "tigera-secure"}
+	if mt {
+		key.Namespace = ns
+	}
+
+	instance := &operatorv1.IntrusionDetection{}
+	err := cli.Get(ctx, key, instance)
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
+
 // Reconcile reads that state of the cluster for a IntrusionDetection object and makes changes based on the state read
 // and what is in the IntrusionDetection.Spec
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	helper := utils.NewNamespaceHelper(r.multiTenant, render.IntrusionDetectionNamespace, request.Namespace)
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
 	reqLogger.Info("Reconciling IntrusionDetection")
 
+	// We skip requests without a namespace specified in multi-tenant setups.
+	if r.multiTenant && request.Namespace == "" {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if this is a tenant-scoped request.
+	tenant, _, err := utils.GetTenant(ctx, r.multiTenant, r.client, request.Namespace)
```

Comment:
I'm a little behind on multi tenancy, but how does this work if a controller needs to reconcile when something changes in a central namespace, like the operator namespace (which I didn't think was multi tenant).

Reply:
In the `Add` function, we use a special even handler called `EneuqueAllTenants` for those resources.

The `Reconcile()` function will be called with the `Tenant` CR instead of the object that changed.